common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/SourceIdentifier.scala

```diff
@@ -6,5 +6,9 @@ case class SourceIdentifier(
   ontologyType: String,
   value: String
 ) {
+  require(
+    value == value.trim,
+    s"SourceIdentifier value contains trailing whitespace: <$value>")
```

Comment:
nit:

```suggestion
    s"SourceIdentifier value contains extra whitespace: <$value>")
```

This assertion could fail if there's leading whitespace, e.g. `" b1234"`